### PR TITLE
Save-compatible SaveBlock3

### PR DESCRIPTION
**Pull request by mrgriffin**
Repurposes the unused space in the footers of save sectors to build a 1624b `SaveBlock3`. Similar to the ideas in [reuse filler save space for variables and flags](https://github.com/pret/pokeemerald/wiki/Reuse-filler-save-space-for-Variables-and-Flags), but with an actual `struct`, and it only occupies as much EWRAM as necessary.

This is non-breaking, so provides a place for us to add new features.

Incompatible with [extra save space with two lines of code](https://github.com/pret/pokeemerald/wiki/Extra-save-space-with-two-lines-of-code), which allocates that additional space to `SaveBlock1` and `SaveBlock2` instead, but it should be relatively straightforward for users to solve the conflicts.

Thanks to Alex and psf for changes to this PR :)

### CODE REVIEW

File: src/load_save.c [hunk 1]
```diff
@@ -57,6 +59,11 @@ void CheckForFlashMemory(void)
     }
 }
 
+void ClearSav3(void)
+{
+  CpuFill16(0, &gSaveblock3, sizeof(struct SaveBlock3));
```
Comment (DizzyEggg) [inline src/load_save.c:64]: tabs

Comment (AlexOn1ine) [reply src/load_save.c]: I took the liberty to fix it. Hope it's fine.

**Comment by Bassoonian** [comment]
As MGriffin mentioned on Discord, this PR is still missing included the space check in the debug menu

**Comment by mrgriffin** [comment]
Ready for review, thanks to psf :)

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**